aspnetcore/blazor/security/passkeys.md

```diff
@@ -0,0 +1,663 @@
+  const optionsResponse = await fetchWithErrorHandling(`/Account/PasskeyRequestOptions?username=${email}`, {
+    method: 'POST',
+    headers,
+    signal,
+  });
+  const optionsJson = await optionsResponse.json();
+  const options = PublicKeyCredential.parseRequestOptionsFromJSON(optionsJson);
+  return await navigator.credentials.get({ publicKey: options, mediation, signal });
+}
+```
+
+The `navigator.credentials.get()` call initiates the authentication process with the authenticator, which prompts the user for verification.
+
+### Step 5: Authenticator verification
+
+The authenticator verifies the user's identity and signs the challenge with the private key. This process is handled entirely by the browser and authenticator, similar to the verification step during registration. The user experience depends on the authenticator type and may involve biometric verification or PIN entry.
+
+### Step 6: Assertion submission
+
+After the authenticator creates the signed assertion, the browser serializes it to JSON and submits it to the server:
+
+```javascript
+async function requestCredential(email, mediation, headers, signal) {
+  // Step 6: The assertion is returned from navigator.credentials.get()
+  // and is serialized to JSON for submission to the server
+  const optionsResponse = await fetchWithErrorHandling(`/Account/PasskeyRequestOptions?username=${email}`, {
+    method: 'POST',
+    headers,
+    signal,
+  });
+  const optionsJson = await optionsResponse.json();
+  const options = PublicKeyCredential.parseRequestOptionsFromJSON(optionsJson);
+  return await navigator.credentials.get({ publicKey: options, mediation, signal });
+}
+```
+
+The submission mechanism varies by app but typically involves either a form submission or an API call.
+
+### Step 7: Server verification
+
+The server verifies the assertion to authenticate the user. ASP.NET Core Identity provides the `PasskeySignInAsync` method, which performs the complete authentication flow in a single call:
+
+```csharp
+var result = await signInManager.PasskeySignInAsync(credentialJson);
+
+if (result.Succeeded)
+{
+    return Results.Ok("Authentication successful");
+}
+
+return Results.Unauthorized();
+```
+
+The `PasskeySignInAsync` method internally calls `PerformPasskeyAssertionAsync` to:
+
+* Validate the assertion signature using the stored public key.
+* Verify that the challenge matches the one originally sent.
+* Check authenticator flags for user presence and verification.
+* Update the signature counter to prevent replay attacks.
+
+If all checks pass, the method signs in the user and returns a `SignInResult` indicating success.
+
+For scenarios requiring more control, you can use `PerformPasskeyAssertionAsync` directly to validate the assertion without immediately signing in the user. This returns a `PasskeyAssertionResult<TUser>` containing the authenticated user and updated passkey information.
```

Comment:
Maybe we could also say that the passkey included in the `PasskeyAssertionResult` should be updated for the user via `userManager.AddOrUpdatePasskeyAsync()`. This is because the passkey's sign count and authenticator flags may have changed since the last assertion.

Reply:
(`PasskeySignInAsync()` automatically stores the updated passkey, so that step is only needed when using `PerformPasskeyAssertionAsync()` directly)